src/rules/authorizedApiGatewaysV2Routes/index.ts

```diff
@@ -0,0 +1,38 @@
+import { Route } from '@aws-sdk/client-apigatewayv2';
+import compact from 'lodash/compact';
+import { fetchAllApiGatewayV2Routes } from '../../aws-sdk-helpers';
+import { GuardianLevel } from '../../constants/level';
+import { Category, Rule } from '../../types';
+
+const isAuthenticated = (route: Route): boolean => {
+  const hasAuthorizer =
+    route.AuthorizationType !== undefined && route.AuthorizationType !== 'NONE';
+
+  return hasAuthorizer;
+};
+
+const run: Rule['run'] = async resourceArns => {
+  const apiGatewaysV2Routes = await fetchAllApiGatewayV2Routes(resourceArns);
+  const results = compact(
+    apiGatewaysV2Routes.flatMap(({ arn, routes }) =>
+      routes?.map(route => ({
+        arn,
+        success: isAuthenticated(route),
+        route: route.RouteKey,
+      })),
+    ),
+  );
+
+  return { results };
+};
+
+const rule: Rule = {
+  ruleName: 'Api Gateway V2: Use Authorized Routes',
+  errorMessage: 'Route is not authorized',
+  run,
+  fileName: 'useAuthorizedRoutes',
+  categories: [Category.SECURITY],
+  level: GuardianLevel.Level5,
```

Comment:
Yes, for two reasons:
* Because this rule can potentially be annoying in the general case if it is normal to have public routes
* Because adding an authorizer to the routes, might potentially mean rethinking the way you handled authentication on your application